Add google cloud

evaldofelipe opened the pull request:
This PR add support to create a base-image on a specific project on GCP.

evaldofelipe committed:
Adjust DO terraform files from past branch

evaldofelipe committed:
Adjust and split documentation

evaldofelipe committed:
Add GCP env exporter to container

evaldofelipe committed:
Add GCP packer provisioner

evaldofelipe committed:
Adjust README.md

katcipis commented on Makefile:
```diff
@@ -30,6 +30,7 @@ base-docker-run = docker run \
 	--env AZURE_TENANT_ID=$(AZURE_TENANT_ID) \
 	--env DO_API_KEY=$(DO_API_KEY) \
```
```suggestion
	--env DO_API_KEY \
```

evaldofelipe replied:
Nice tip! thanks!

katcipis commented on README.md:
```diff
@@ -48,53 +20,28 @@ Setup the docker image with [Terraform](https://www.terraform.io/) and [Packer](
 
-images
- - image-ubuntu
+On this project we separate the multi cloud schema and tools, with arguments on execution of Makefile. The valid arguments are:
```
```suggestion
On this project we separate the multi cloud schema and tools using arguments when executing make. The valid arguments are:
```

katcipis commented on docs/azure.md:
````diff
@@ -0,0 +1,43 @@
+### Initialize the packer environment
+
+```bash
+$ make terraform-init provider=azure env=images-builder
````
I'm not aware of details on the project etc, but it seems a little off that the description says that you are going to initialize the packer environment but the make target is "make terraform-init". It seems like some leaking of the internal tool used to do stuff. If i'm initializing a packer env I would expect "make packer-env" (terraform is the detail). But it is just some food for thought, perhaps there is good reasoning to keep it this way.

evaldofelipe replied:
Thanks for the advice! I'll explain trough a better way what we doing here.

katcipis commented on docs/azure.md:
````diff
@@ -0,0 +1,43 @@
+### Create the packer environment
+
+```bash
+$ make terraform-apply provider=azure env=images-builder
````
Same thing as before. it seems like the tool details leak and the target is not informing me of what is actually being done (creating a packer env). The core concept is, could the make targets be written on a way that even without docs I would understand what is happening ? Not that this way is impossible to understand, but from the perspective of someone that does not know the project very well it seems a little hard to understand.

Another thing to think is the leverage that another layer like makes gives you. You are already abstracting how things are actually being built, I just want to build them. It would be like compiling C code with make build-gcc, usually the compiler is abstracted away in favor of what you are trying to build.

katcipis commented on docs/gcp.md:
````diff
@@ -0,0 +1,25 @@
+$ make packer-build env=google-cloud image=image-ubuntu
+```
+
+Afer this step, you will have a image on your image list repository, inside the project where your Service Account Key was generated.
````
```suggestion
After this step, you will have a image on your image list repository, inside the project where your Service Account Key was generated.
```

katcipis commented on packer/builders/google-cloud/image-ubuntu/provisioner.json:
```diff
@@ -0,0 +1,26 @@
+    {
+      "type": "googlecompute",
+      "account_file": "../account.json",
+      "project_id": "blackops-qa",
```
is it OK to have this hardcoded here ? (I'm not even sure what is this JSON)

evaldofelipe replied:
I'm thinking in a better way to store this, but at this moment we are storing this file `account.json` inside on container as variable and writing in this file to use to access the provider resources.

katcipis commented on terraform/providers/digital-ocean/images-tester/tester-environment.tf:
```diff
@@ -22,11 +22,10 @@ resource "digitalocean_droplet" "droplet" {
-      timeout = "2m"
+    user        = "packer"
+    type        = "ssh"
+    private_key = "${file("~/.ssh/id_rsa")}"
```
The key file is taken from the HOME of the host or the container ?

evaldofelipe replied:
yeah, this is a sskey is copied from localhost default path to container. but is can be documented too. Thanks for this advice! 👌

evaldofelipe replied:
Open a issue #19 to review the real need of the local ssh key on entire project.

katcipis reviewed (dismissed):
Minor typos and some ideas on target naming that should probably be ignored =P

evaldofelipe committed:
GCP documentation and makefile vars adjustments

evaldofelipe committed:
Adjust azure documentation

evaldofelipe committed:
Fix documentation

evaldofelipe committed:
Add travis env variable for image hash and change auth json path

evaldofelipe committed:
Add gcp key to ignore

evaldofelipe committed:
Add specific vars to use ansible role

evaldofelipe committed:
Add neoway ansible role repository

evaldofelipe committed:
Add gcp to travis

evaldofelipe committed:
Add terraform provider for google tester instance

evaldofelipe committed:
Add entrypoint hack to container

evaldofelipe committed:
Add new env to travis

evaldofelipe committed:
Remove playbook

evaldofelipe committed:
Adjust json output

evaldofelipe committed:
Fix variable echo to jq

evaldofelipe committed:
Edit ubuntu image version

evaldofelipe committed:
Temporary env key test

evaldofelipe committed:
Add new variable for travis

evaldofelipe committed:
Remove file test validation

evaldofelipe committed:
Adjust travis variable

evaldofelipe committed:
Add travis variable to docker run

evaldofelipe committed:
Change base64 output command

evaldofelipe committed:
Adjust output travis variables

lborguetti committed:
terraform->gcp->images-tester: Change to use credentials_file

    Terraform dosen't able to authenticate to GCP and fail to create
    resources.
    This commit change the way to authenticate using credentials_file.

lborguetti committed:
hack->entrypoint.sh: fix bash syntax error

lborguetti committed:
hack->entrypoint.sh: fix gcp_token lowercase

lborguetti committed:
Makefile: Add TRAVIS_BUILD_ID

evaldofelipe committed:
Add random variable to VM name

evaldofelipe committed:
Pull image from dockerhub for travis

evaldofelipe committed:
Build container on Travis

evaldofelipe committed:
Adjust random_id for VM name

lborguetti reviewed (approved):
LGTM!
